test/e2e-shared-tests/helpers.go

```diff
@@ -53,16 +53,34 @@ func waitForConfigDriftMonitorStart(t *testing.T, cs *framework.ClientSet, node
 
 	start := time.Now()
 
+	ctx := context.Background()
+
 	err := wait.PollImmediate(2*time.Second, 1*time.Minute, func() (bool, error) {
 		mcdPod, err := helpers.MCDForNode(cs, &node)
 		require.Nil(t, err)
 
 		logs, err := cs.Pods(mcdPod.Namespace).GetLogs(mcdPod.Name, &corev1.PodLogOptions{
 			Container: "machine-config-daemon",
-		}).DoRaw(context.TODO())
+		}).DoRaw(ctx)
 		require.Nil(t, err)
 
-		return strings.Contains(string(logs), configDriftMonitorStartupMsg), nil
+		splitLogs := strings.Split(string(logs), "\n")
+
+		// Scan the logs from the bottom up, looking for either a shutdown or startup message.
+		for i := len(splitLogs) - 1; i != 0; i-- {
```

Comment:
I think this is problematic if the logs if empty. We can probably work around it with `i<=0` though

Reply:
Good catch, thanks!